core/src/Http/HttpRequest.cs

```diff
@@ -46,10 +46,16 @@ private HttpRequest()
 
         public static async Task<HttpResponse> SendPost(Uri endpoint, Dictionary<string, string> headers,
             Dictionary<string, string> bodyParameters, RequestContext requestContext)
+        {
+            return await SendPost(endpoint, headers, new FormUrlEncodedContent(bodyParameters), requestContext).ConfigureAwait(false);
```

Comment:
What is bodyParameters is null? There is such a check below.

Reply:
Good catch! I am adding a check here. We don't need an empty dictionary though, a null pointer would be enough.

            HttpContent body = null;
            if (bodyParameters != null)
            {
                body = new FormUrlEncodedContent(bodyParameters);
            }

Reply:
I answered this days ago in a github comment, which was not showed as a reply to your original comment in CodeFlow. So I'm copying that reply here:

Good catch! I am adding a check here. We don't need an empty dictionary though, a null pointer would be enough.

            HttpContent body = null;
            if (bodyParameters != null)
            {
                body = new FormUrlEncodedContent(bodyParameters);
            }

---
In reply to: [203669701](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r203669701) [](ancestors = 203669701)